exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -191,12 +189,12 @@ sdk::common::ExportResult ElasticsearchLogExporter::Export(
   std::string responseBody = handler->GetResponseBody();
   if (responseBody.find("\"failed\" : 0") == std::string::npos)
   {
-    if (options_.console_debug_)
-    {
-      std::cout << "Logs were not written to Elasticsearch correctly, response body:" << std::endl;
-      std::cout << responseBody << std::endl;
-    }
-
+    std::stringstream ss;
```

Comment:
One issue with a macro that accepts only `std::string` or `const char *`, and not vararg parameters:
- you lose the knowledge of concrete strongly typed values by blending them all into one single string (so that your callback inherently loses all relevant details, e.g. you can't pass a NUMBER or ADDRESS to the callback).

Value of having strongly-typed parameters in logging: you set a breakpoint, and/or even conditional breakpoint to catch when param equals to certain value. For example, `OTEL_LOG_ERROR("http status code=%d", 403);` - see, that way your callback actually knows that it's 403 and does not need to parse. I would've also added an overload to the `OTEL_LOG_ERROR` function to pass context info, such as done here:
https://github.com/microsoft/cpp_client_telemetry/blob/7dd687c71297e1c72f6126896662f9c1e70d690f/lib/include/public/DebugEvents.hpp#L137

You can pass almost an entire call stack knowledge using a class like this:

```cpp
    /// <summary>The DebugEvent class represents a debug event object.</summary>
    class DebugEvent
    {

    public:
        /// <summary>The debug event sequence number.</summary>
        uint64_t seq;
        /// <summary>The debug event timestamp.</summary>
        uint64_t ts;
        /// <summary>The debug event type.</summary>
        DebugEventType type;
        /// <summary>[optional] Parameter 1 (depends on debug event type).</summary>
        size_t param1;
        /// <summary>[optional] Parameter 2 (depends on debug event type).</summary>
        size_t param2;
        /// <summary>[optional] The debug event data (depends on debug event type).</summary>
        void* data;
        /// <summary>[optional] The size of the debug event data (depends on debug event type).</summary>
        size_t size;

        /// <summary>DebugEvent The default DebugEvent constructor.</summary>
        DebugEvent() : seq(0), ts(0), type(EVT_UNKNOWN), param1(0), param2(0), data(NULL), size(0) {};

        DebugEvent(DebugEventType type) : seq(0), ts(0), type(type), param1(0), param2(0), data(NULL), size(0) {};

        DebugEvent(DebugEventType type, size_t param1, size_t param2 = 0, void* data = nullptr, size_t size = 0) :
            seq(0), ts(0), type(type), param1(param1), param2(param2), data(data), size(size) {};
    };
```

You can borrow this model entirely. It's Apache License v2.0. Works great elsewhere. 😄